### PR TITLE
Add tests to check creation of 1D components and fix a few bugs.

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Add tests to check the creation of 1D components,
- [x] fix a few issues,
- [x] ready for review.

### CODE REVIEW

File: hyperspy/tests/component/test_components.py [hunk 1]
```diff
@@ -24,11 +24,39 @@
...
 from hyperspy.misc.test_utils import ignore_warning
+from hyperspy import components1d
 
 
 TRUE_FALSE_2_TUPLE = [p for p in itertools.product((True, False), repeat=2)]
 
 
+def get_components1d_list():
+    components1d_list = [c_name for c_name in
+                         dir(components1d) if '_' not in c_name]
```
Comment (thomasaarholt) [inline hyperspy/tests/component/test_components.py]: The following will still do all tests if a component name ends up containing an underscore:
```suggestion
                         dir(components1d) if not cname.startswith('_')]
```

**Review by thomasaarholt — COMMENTED** [review]
Only one comment, otherwise LGTM!

File: hyperspy/tests/component/test_components.py (hunk 1 above)
Comment (francisco-dlp) [reply hyperspy/tests/component/test_components.py]: @ericpre, are you planning to take into account @thomasaarholt comment?

Comment (ericpre) [reply hyperspy/tests/component/test_components.py]: Following the current convention, the name of the components should not contain underscore. Anyway this was not a good way to do it so I changed it to be more reliable.

**Pull request merged**